python/housinginsights/sources/wmata_distcalc.py

```diff
@@ -0,0 +1,234 @@
+        time.sleep(0.8)
+        if "Too Many Requests" in str(walkDistResponse.json()):
+                i = 0
+                while "Too Many Requests" in str(walkDistResponse.json()) and i < 10:
+                    walkDistResponse = requests.get("https://api.mapbox.com/directions/v5/mapbox/walking/" + distReqCoords,params=mapbox_params)
+                    i = i + 1
+                    if i == 10:
+                        raise Exception('This is some exception to be defined later')
+        return walkDistResponse.json()['routes'][0]['legs'][0]['distance']
+
+
+    def setProjectInfo(self,project,columns):
+        for column_name in columns:
+            if 'proj_lat' == column_name[0]:
+                self.lat = project[columns.index(column_name)]
+            elif 'proj_lon' == column_name[0]:
+                self.lon = project[columns.index(column_name)]
+            elif 'nlihc_id' == column_name[0]:
+                self.nlihcid = project[columns.index(column_name)]
+
+    def findRailStations(self, railStations,project,radiusinmeters,distCsvWriter):
+        """Finds all the rail stations within a given distance from a given project.  Writes to the given CSV file.
+
+        Parameters:
+        railStations - json object containing all the wmata rail station information
+        project - housing project object
+        radiusinmeters - radius in meteres
+        distCsvWriter - csvWriter for distance
+        mapbox_api_key - api key for mapbox REST services
+        """
+
+        lat = self.lat
+        lon = self.lon
+        Nlihc_id = self.nlihcid
+
+        for station in railStations:
+            walkDist = self.getWalkingDistance(lat, lon, str(station['Lat']), str(station['Lon']))
+
+            if walkDist <=radiusinmeters:
+                #railData.append([Nlihc_id, 'rail', station['Code'], "{0:.2f}".format(self.miles)])
+                distCsvWriter.writerow([Nlihc_id, 'rail', station['Code'], "{0:.2f}".format(self.miles)])
+
+        #self.result_to_csv(distHeader,railData,self.distOutputFileName)
+
+    def findBusStations(self, project,radiusinmeters, distCsvWriter):
+        lat = self.lat
+        lon = self.lon
+        Nlihc_id = self.nlihcid
+
+        wmata_headers = self.getWmataHeaders()
+
+        params = {'Lat': lat,
+                  'Lon' : lon,
+                  'Radius':str(radiusinmeters)}
+        response = requests.get('https://api.wmata.com/Bus.svc/json/jStops', params=params, headers=wmata_headers)
+        data = response.json()
+
+        for stop in data['Stops']:
+            walkDist = self.getWalkingDistance(lat, lon, str(stop['Lat']), str(stop['Lon']))
+            if walkDist <= radiusinmeters: #within 0.5 miles walking
+                distCsvWriter.writerow([Nlihc_id, 'bus', stop['StopID'], "{0:.2f}".format(self.miles)])
+
+    def writeRailInfo(self, infoCsvWriter):
```

Comment:
Maybe we can rename this to get_all_rail_stations? That appears to be what it's doing, right?